src/OpenDiffix.Core/QueryEngine.fs

```diff
@@ -15,6 +15,32 @@ let rec private extractColumns query =
 
 type QueryResult = { Columns: Column list; Rows: Row list }
 
+// The main entrypoint to the `OpenDiffix.Core` library.
+// Takes in a SQL `statement` as main parameter along with a set of
+// parameters in `queryContext`.
+//
+// Returns the query result as a list of data rows, along with the
+// columns specifications.
+//
+// Supports two main modes of operation:
+// I. Standard SQL statement, with `aidColumns` specified within the `queryContext`
+//
+// Will rewrite the statement so that the query result is anonymized according to the
+// `aidColumns` provided. For example, it will append `AND NOT diffix_low_count(aidColumns)`
+// to the `HAVING` SQL clause and rewrite regural SQL aggregators to Diffix aggregators
```

Comment:
regural -> regular

Reply:
hm, just noticed vscode spellchecker extension doesn't work for `.fs` files, sorry about that